Tasks/google-play-release/GooglePlay.ts

```diff
@@ -768,30 +773,51 @@ function resolveGlobPaths(path: string): string[] {
     return [];
 }
 
+function getObbFromParentDirectroy(apkPath: string): string {
+    let obbFile: string = null;
+    let apkDirectory = path.dirname(apkPath);
+    let parentDirectory = path.dirname(apkDirectory);
+    let filenames = fs.readdirSync(parentDirectory);
+    filenames.forEach(file => {
+        if (path.extname(file) === '.obb') {
+            obbFile = file;
+        }
+    });
+
+    if (obbFile !== null) {
+        return path.join(parentDirectory, obbFile);
+    } else {
+        tl.debug(`No Obb found for ${apkPath}, skipping upload`);
+        return obbFile;
+    }
+}
+
 /**
  * Get unique APK file paths from main and additional APK file inputs.
  * @returns {string[]} paths of the files
  */
-async function getAllApkPaths(mainApkFile: string): Promise<string[]> {
-    const apkFileList: { [key: string]: number } = {};
+async function getAllApkPaths(mainApkFile: string): Promise<Object> {
```

Comment:
I'm not sure that this is the best way to implement this.  Returning an `Object` is usually a sign that you're trying to do too much with a method.

I think this would be clearer to have this and `getObbFromParentDirectroy` remain as-is, but line ~116 would check `shouldPickObbFile` and call `getObbFromParentDirectroy` as needed.

Is there a reason why you need to do it here?  I don't think performance would matter.  It should be the same if it were above.  Any other reason?

Reply:
So I took this approach, because in that function, a dictionary was used to hold apkFile paths as key and a number as value. But these numbers were all assigned zero and I didn't exactly get the reason for this. So I though I would just change those values to assign obb paths. So should I go with this approach or the one you suggested?

Reply:
Hrmm... I see your point.  The original code is a little wonky too.  I still feel my suggestion would make the code more maintainable and readable.  But, if you are disinclined to make the change, then I'd be happy with a strongly typed returned value for the function with a comment describing it.  Thanks!

Reply:
Yes, I would like to take your approach. But there is one more issue here, the shouldPickObbFile and shouldPickObbFileForAdditionalPaths are kept as two separate options and we need to check which one is checked to upload obb correspondingly. Should I just merge them into a single option and pick obb files for main as well as additionalApk if available? (in the helpMarkDown we mention that the option if checked would pick obb for both main and additional apk if available, else would skip it). What do you suggest?